From 8eeeadbc31e7c6097be537924d641c489e7f7b0f Mon Sep 17 00:00:00 2001
From: Meson Codec <meson-codec@localhost>
Date: Sun, 23 Aug 2026 11:18:37 +0800
Subject: [PATCH] arm64: dts: meson-gx: describe HCODEC encoder resources

Add the vendor-documented HCODEC mailbox interrupt and connect the HCODEC
clock on GXL. GXM inherits the GXL clock list and the common GX interrupt.

Signed-off-by: Meson Codec <meson-codec@localhost>
---
 arch/arm64/boot/dts/amlogic/meson-gx.dtsi  | 5 +++--
 arch/arm64/boot/dts/amlogic/meson-gxl.dtsi | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
index 0fb2e06..594a204 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gx.dtsi
@@ -555,8 +555,9 @@ vdec: video-codec@c8820000 {
 			reg-names = "dos", "esparser";
 
 			interrupts = <GIC_SPI 44 IRQ_TYPE_EDGE_RISING>,
-				     <GIC_SPI 32 IRQ_TYPE_EDGE_RISING>;
-			interrupt-names = "vdec", "esparser";
+				     <GIC_SPI 32 IRQ_TYPE_EDGE_RISING>,
+				     <GIC_SPI 45 IRQ_TYPE_EDGE_RISING>;
+			interrupt-names = "vdec", "esparser", "venc";
 
 			amlogic,ao-sysctrl = <&sysctrl_AO>;
 			amlogic,canvas = <&canvas>;
diff --git a/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi b/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi
index e202d84..fb50331 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl.dtsi
@@ -973,8 +973,9 @@ &vdec {
 	clocks = <&clkc CLKID_DOS_PARSER>,
 		 <&clkc CLKID_DOS>,
 		 <&clkc CLKID_VDEC_1>,
-		 <&clkc CLKID_VDEC_HEVC>;
-	clock-names = "dos_parser", "dos", "vdec_1", "vdec_hevc";
+		 <&clkc CLKID_VDEC_HEVC>,
+		 <&clkc CLKID_VDEC_HCODEC>;
+	clock-names = "dos_parser", "dos", "vdec_1", "vdec_hevc", "hcodec";
 	resets = <&reset RESET_PARSER>;
 	reset-names = "esparser";
 };
-- 
2.47.3
